justfile

```
@@ -0,0 +1,56 @@
+alias b := build
+alias c := check
+alias t := test
+
+default:
+  @just --list
+
+# run `cargo build` on everything
+build *ARGS="--workspace --all-targets":
+  #!/usr/bin/env bash
+  set -euo pipefail
+  if [ ! -f Cargo.toml ]; then
+    cd {{invocation_directory()}}
+  fi
+  cargo build {{ARGS}}
+
+# run `cargo check` on everything
+check *ARGS="--workspace --all-targets":
+  #!/usr/bin/env bash
+  set -euo pipefail
+  if [ ! -f Cargo.toml ]; then
+    cd {{invocation_directory()}}
+  fi
+  cargo check {{ARGS}}
+
+# run code formatters
+format:
+  #!/usr/bin/env bash
+  set -euo pipefail
+  if [ ! -f Cargo.toml ]; then
+    cd {{invocation_directory()}}
+  fi
+  cargo fmt --all
+  nixpkgs-fmt $(echo **.nix)
+
+# run tests
+test: build
+  #!/usr/bin/env bash
+  set -euo pipefail
+  if [ ! -f Cargo.toml ]; then
+    cd {{invocation_directory()}}
+  fi
+  cargo test
```

You shouldn't need to call `build` first but you do need the `ARGS`.

```suggestion
test *ARGS="--workspace --all-targets":
  #!/usr/bin/env bash
  set -euo pipefail
  if [ ! -f Cargo.toml ]; then
    cd {{invocation_directory()}}
  fi
  cargo test {{ARGS}}
```

---

Done! 0267841eef6243516bcdc95da0d1cb777fb687c5